src/bilayers/build/parse/cellprofiler_plugin_template.py.j2

```diff
@@ -0,0 +1,655 @@
+
+Technical notes
+^^^^^^^^^^^^^^^
+
+This module runs the {{ docker_image.get('name') }} algorithm in a container (Docker or Podman).
+The container image used is: {{ docker_image.get('org') }}/{{ docker_image.get('name') }}:{{ docker_image.get('tag') }}
+
+{% if citations %}
+References
+^^^^^^^^^^
+
+{% for citation in citations.values() %}
+- {{ citation.name }}: {{ citation.description }}
+  DOI: {{ citation.doi }}
+  License: {{ citation.license }}
+{% endfor %}
+{% endif %}
+"""
+
+class {{ algorithm_folder_name }}(Module):
+    #
+    # The module starts by declaring the name that's used for display,
+    # the category under which it is stored and the variable revision
+    # number which can be used to provide backwards compatibility if
+    # you add user-interface functionality later.
+    #
+    {% if computed_category == "Image Segmentation" %}
+    category = "Object Processing"
+    {% else %}
+    category = "{{ computed_category }}"
+    {% endif %}
+    
+    module_name = "{{ algorithm_folder_name }}"
+
+    variable_revision_number = 1
+
+    #
+    # Citation - Please cite the following when using this module
+    #
+    {% if citations %}
+    doi = {
+        {% for citation in citations.values() %}
+        "Please cite {{ citation.name }} when using this module": "{{ citation.doi }}",
+        {% endfor %}
+    }
+    {% endif %}
+
+    #
+    # "create_settings" is where you declare the user interface elements
+    # (the "settings") which the user will use to customize your module.
+    #
+    def create_settings(self):
+        {% set type_mapping = {
+            "image": "ImageName",
+            "measurement": "Filename",
+            "array": "Filename",
+            "executable": "Filename",
+            "file": "Filename",
+            "radio": "Choice",
+            "dropdown": "Choice",
+            "checkbox": "Binary",
+            "integer": "Integer",
+            "float": "Float",
+            "textbox": "Text",
+        } %}
+        {% set default_type = "Text" %} {# Fallback type #}
+
+        # Create input settings for all modules
+        {% for key, input_conf in inputs.items() %}
+        {% if input_conf.get('type') == 'image' %}
+        self.{{ input_conf.get('name') }} = ImageSubscriber(
+            "{{ input_conf.get('label') }}",
+            doc="""\
+{{ input_conf.get('description') }}
+"""
+        )
+        {% else %}
+        self.{{ input_conf.get('name') }} = {{ type_mapping.get(input_conf.get('type'), default_type) }}(
+            text="{{ input_conf.get('label') }}",
+            {% if input_conf.get('default') is not none %}
+            value="{{ input_conf.get('default') }}",
+            {% endif %}
+            doc="""\
+{{ input_conf.get('description') }}
+"""
+        )
+        {% endif %}
+        {% endfor %}
+
+        # Create output settings for all modules
+        {% for key, output_conf in outputs.items() %}
+        {% if output_conf.get('type') == 'image' %}
```

Comment:
(I know you know this, just marking open to-dos)- TODO: check if the image subtype is label, if so, use the label namer